core-bundle/tests/Cron/CronTest.php

```diff
@@ -0,0 +1,174 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Tests\Cron;
+
+use Contao\CoreBundle\Cron\Cron;
+use Contao\CoreBundle\Entity\CronJob;
+use Contao\CoreBundle\Fixtures\Cron\TestInvokableScopedCronJob;
+use Contao\CoreBundle\Repository\CronJobRepository;
+use Contao\CoreBundle\Tests\TestCase;
+use PHPUnit\Framework\MockObject\MockObject;
+use Psr\Log\LoggerInterface;
+
+class CronTest extends TestCase
+{
+    public function testExecutesAddedCronJob(): void
+    {
+        $repository = $this->createMock(CronJobRepository::class);
+
+        $cronjob = $this->getMockBuilder(\stdClass::class)->addMethods(['onHourly'])->getMock();
```

Comment:
Why are you mocking a `stdClass` object here? Which object is run in the real application later?

Reply:
An arbitrary object. Bug I guess there is probably a better way? We can also use the test cronjob classes from the fixtures.

Reply:
What about `$this->createMock(ScopedCronJobInterface::class)`?

Reply:
That will not be enough, as I need the cron job methods as well. I cannot add additional methods of a mock object of an existing class, can I? Only via the mock builder, afaik.

Reply:
Then let's use the fixtures class.

Reply:
Another problem I am running into here: when I use

```php
$cronjob = $this->createMock(TestCronJob::class);
```

I cannot test the logging completely, as the mocked class' name will be something random (e.g. `Mock_TestCronJob_a0f1e286`) - while when using

```php
 $cronjob = $this
    ->getMockBuilder(\stdClass::class)
    ->setMockClassName('TestCronJob')
    ->addMethods(['onMinutely', 'onHourly'])
    ->getMock()
;
```

it will be exactly what I set it to (i.e. `TestCronJob`).

Reply:
Nevertheless, I adjusted the tests.

Reply:
Why do you mock the fixtures class at all? It already is a fixtures class, so you can use it directly.

Reply:
Right 😁